src/server.rs

```diff
@@ -368,6 +374,50 @@ mod tests {
         Ok(())
     }
 
+    #[actix_web::test]
+    async fn test_index_with_landing_page_file() -> Result<(), Error> {
+        let filename = "landing_page.txt";
+        let mut config = Config::default();
+        let mut file = File::create(filename)?;
+        file.write_all("landing page from file".as_bytes())?;
+        config.server.landing_page_file = Some(filename.to_string());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(RwLock::new(config)))
+                .service(index),
+        )
+        .await;
+        let request = TestRequest::default()
+            .insert_header(("content-type", "text/plain"))
+            .to_request();
+        let response = test::call_service(&app, request).await;
+        assert_eq!(StatusCode::OK, response.status());
+        assert_body(response.into_body(), "landing page from file").await?;
+        fs::remove_file(filename)?;
+        Ok(())
+    }
+
+    #[actix_web::test]
+    async fn test_index_with_landing_page_file_not_found() -> Result<(), Error> {
+        let filename = "landing_page.txt";
+        let mut config = Config::default();
+        config.server.landing_page = Some(String::from("landing page"));
+        config.server.landing_page_file = Some(filename.to_string());
+        let app = test::init_service(
+            App::new()
+                .app_data(Data::new(RwLock::new(config)))
+                .service(index),
+        )
+        .await;
+        let request = TestRequest::default()
+            .insert_header(("content-type", "text/plain"))
+            .to_request();
+        let response = test::call_service(&app, request).await;
+        assert_eq!(StatusCode::FOUND, response.status());
+        //assert_body(response.into_body(), "landing page").await?;
```

Comment:
Can you remove this comment?

Reply:
done.

I only kept it in case you wanted the fallback to `landing_page`....